proxy/hdrs/URL.cc

```diff
@@ -1588,10 +1594,33 @@ url_print(URLImpl *url, char *buf_start, int buf_length, int *buf_index_inout, i
   if (!x)      \
   return 0
 
+  enum { OTHER, FILE, NONE } scheme_added = OTHER;
   if (url->m_ptr_scheme) {
-    TRY((normalize ? mime_mem_print_lc : mime_mem_print)(url->m_ptr_scheme, url->m_len_scheme, buf_start, buf_length,
-                                                         buf_index_inout, buf_chars_to_skip_inout));
-    TRY(mime_mem_print("://", 3, buf_start, buf_length, buf_index_inout, buf_chars_to_skip_inout));
+    if ((url->m_scheme_wks_idx >= 0) && (hdrtoken_index_to_wks(url->m_scheme_wks_idx) == URL_SCHEME_FILE)) {
+      TRY(mime_mem_print(URL_SCHEME_FILE, URL_LEN_FILE, buf_start, buf_length, buf_index_inout, buf_chars_to_skip_inout));
+      scheme_added = FILE;
```

Comment:
As a bit of unrelated cleanup, allow for "file" scheme, since url_length_get() allows for it.